File: server/src/protocol/GuardStatusCommandBuilder.cpp
#include "GuardStatusBuilder.h"
#include "GuardStatusCommand.h"

using namespace std;
using namespace communication;

GuardStatusBuilder::GuardStatusBuilder()
{}

GuardStatusBuilder::~GuardStatusBuilder()
{}

unique_ptr<Command> GuardStatusBuilder::create(const vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<GuardStatusCommand>();

    return move(command);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/GuardStatusCommandBuilder.java
package wpam.mobile_client.protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class GuardStatusCommandBuilder implements CommandBuilder, Closeable
{
	public GuardStatusCommandBuilder()
	{
	}
	public final void close()
	{
	}

	@Override
	public Command create(ArrayList<Integer> commandInBytes)
	{
		Command command  = new GuardStatusCommand();
		return command;
	}
}